main.py

```diff
@@ -949,19 +944,34 @@ def on_cleanup_evcs_old_path(self, event):
     def handle_cleanup_evcs_old_path(self, event):
         """Handle cleanup evcs old path."""
         evcs = event.content.get("evcs", [])
-        event_contents: dict[str, list[dict]] = {}
+        event_contents: dict[str, dict] = defaultdict(list)
+        total_flows = {}
         for evc in evcs:
             if not evc.old_path:
                 continue
-            removed_flows = evc.remove_path_flows(evc.old_path)
-            content = map_evc_event_content(
-                evc,
-                removed_flows=removed_flows,
-                current_path=evc.current_path.as_dict(),
-            )
-            event_contents[evc.id] = content
-            evc.old_path = Path([])
+            with evc.lock:
+                removed_flows = {}
+                try:
+                    nni_flows = evc._prepare_nni_flows(evc.old_path)
+                    uni_flows = evc._prepare_uni_flows(
+                        evc.old_path, skip_in=True
+                    )
+                    removed_flows = merge_flow_dicts(nni_flows, uni_flows)
+                # pylint: disable=broad-except
+                except Exception:
+                    err = traceback.format_exc().replace("\n", ", ")
+                    log.error(f"Fail to remove {evc} old_path: {err}")
+                    continue
+            if removed_flows:
+                total_flows = merge_flow_dicts(removed_flows, total_flows)
+                content = map_evc_event_content(
+                    evc,
+                    removed_flows=removed_flows,
+                    current_path=evc.current_path.as_dict(),
+                )
+                event_contents[evc.id] = content
         if event_contents:
+            send_flow_mods_event(self.controller, total_flows, 'delete')
```

Comment:
@Alopalao, nice improvement here not to have a failable request and send via events.  You can even make this more reliable during a `deletion` if you were to set and parametrize the force=True value (which the request deletion also uses), that way flow_manager deals handling temporarily switch losing connectivity too. Can you set here and augment `send_flow_mods_event` accordingly?